### PR TITLE
Add `warnOnNonMainThreadDetected` configuration to `IOApp`

### PR DESCRIPTION
**Pull request by diogocanut**
Solves #3806

### CODE REVIEW

**Comment by armanbilge** [comment]
Thanks, but this is missing the other changes from https://github.com/typelevel/cats-effect/pull/3814 where you actually use `warnOnNonMainThreadDetected` :) everything you did there was perfect, we just need to target it at this branch.

**Comment by armanbilge** [comment]
Oh of course 🤦 that's why you couldn't make the change, because we didn't merge `series/3.5.x` branch into `series/3.x` branch yet. Sorry I forgot about that. We should wait until that happens, then this PR makes sense.

**Comment by armanbilge** [comment]
@diogocanut we just merged `series/3.5.x` into `series/3.x` so you should be able to finish this PR now. Sorry for my mistake before!

**Comment by diogocanut** [comment]
@armanbilge done, thanks for noticing.

File: core/jvm/src/main/scala/cats/effect/IOApp.scala [hunk 1]
```diff
@@ -337,12 +337,14 @@ trait IOApp {
    * isn't the main process thread. This condition can happen when we are running inside of an
    * `sbt run` with `fork := false`
    */
+
```
Comment (armanbilge) [inline core/jvm/src/main/scala/cats/effect/IOApp.scala]: Stray space?
```suggestion
```

Comment (diogocanut) [reply core/jvm/src/main/scala/cats/effect/IOApp.scala]: ops, removed.

**Comment by Daenyth** [comment]
bump

**Review by armanbilge — APPROVED** [review]

**Pull request merged**